### PR TITLE
cilium: configurable ct timeouts

### PR DESCRIPTION
**Pull request by borkmann**
(See commit msgs.)

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/9124)

### CODE REVIEW

**Comment by borkmann** [comment]
test-me-please

File: pkg/option/config.go [hunk 1]
```diff
@@ -387,6 +387,20 @@ const (
...
+	// CTMapEntriesTimeout* name option and default value mappings
+	CTMapEntriesTimeoutSYNName       = "bpf-ct-timeout-regular-tcp-syn"
+	CTMapEntriesTimeoutSYNDefault    = "1m"
+	CTMapEntriesTimeoutFINName       = "bpf-ct-timeout-regular-tcp-fin"
+	CTMapEntriesTimeoutFINDefault    = "10s"
+	CTMapEntriesTimeoutTCPName       = "bpf-ct-timeout-regular-tcp"
+	CTMapEntriesTimeoutTCPDefault    = "6h"
+	CTMapEntriesTimeoutAnyName       = "bpf-ct-timeout-regular-any"
+	CTMapEntriesTimeoutAnyDefault    = "1m"
+	CTMapEntriesTimeoutSrvTCPName    = "bpf-ct-timeout-service-tcp"
```
Comment (brb) [inline pkg/option/config.go]: Nit: maybe `s/CTMapEntriesTimeoutSrvTCPName/CTMapEntriesTimeoutSVCTCPName/`, as in the cilium-agent codebase we abbreviate "service" with "svc"?

**Comment by borkmann** [comment]
test-me-please

**Review by brb — APPROVED** [review]
LGTM, just one nit.

File: pkg/option/config.go (hunk 1 above)
Comment (borkmann) [reply pkg/option/config.go]: Ok, I can do that.

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
Unrelated CI issue:
```
13:13:39      k8s1-1.15: Get:35 http://security.ubuntu.com/ubuntu bionic-security/universe Translation-en [201 kB]
13:13:39      k8s1-1.15: Get:36 http://security.ubuntu.com/ubuntu bionic-security/multiverse i386 Packages [4,148 B]
13:13:39      k8s1-1.15: Get:37 http://security.ubuntu.com/ubuntu bionic-security/multiverse amd64 Packages [4,688 B]
13:13:39      k8s1-1.15: Get:38 http://security.ubuntu.com/ubuntu bionic-security/multiverse Translation-en [2,356 B]
13:13:45      k8s1-1.15: Fetched 2,180 kB in 16s (136 kB/s)
13:13:45      k8s1-1.15: Reading package lists...
13:13:46      k8s1-1.15: Reading package lists...
13:13:46      k8s1-1.15: Building dependency tree...
13:13:46      k8s1-1.15: Reading state information...
13:13:46      k8s1-1.15: No apt package "kubeadm", but there is a snap with that name.
13:13:46      k8s1-1.15: Try "snap install kubeadm"
13:13:46      k8s1-1.15: No apt package "kubectl", but there is a snap with that name.
13:13:46      k8s1-1.15: Try "snap install kubectl"
13:13:46      k8s1-1.15: No apt package "kubelet", but there is a snap with that name.
13:13:46      k8s1-1.15: Try "snap install kubelet"
13:13:46      k8s1-1.15: E
13:13:46      k8s1-1.15: : 
13:13:46      k8s1-1.15: Unable to locate package kubernetes-cni
13:13:46      k8s1-1.15: E
13:13:46      k8s1-1.15: : 
13:13:46      k8s1-1.15: Unable to locate package kubelet
13:13:46      k8s1-1.15: 
13:13:46      k8s1-1.15: E: Unable to locate package kubeadm
13:13:46      k8s1-1.15: E: Unable to locate package kubectl
13:13:46  The SSH command responded with a non-zero exit status. Vagrant
```

File: daemon/daemon_main.go [hunk 2]
```diff
@@ -599,6 +599,24 @@ func init() {
 	flags.Int(option.CTMapEntriesGlobalAnyName, option.CTMapEntriesGlobalAnyDefault, "Maximum number of entries in non-TCP CT table")
 	option.BindEnvWithLegacyEnvFallback(option.CTMapEntriesGlobalAnyName, "CILIUM_GLOBAL_CT_MAX_ANY")
 
+	flags.Duration(option.CTMapEntriesTimeoutTCPName, 21600*time.Second, "Timeout for established entries in TCP CT table")
```
Comment (aanm) [inline daemon/daemon_main.go:602]: @borkmann for readability this can be `6*time.Hour` and `time.Minute` for the `60*time.Second`

Comment (borkmann) [reply daemon/daemon_main.go:602]: True, I left it as seconds here to have a better comparison wrt the defines from C which got removed.

**Pull request merged**